Convert to a WebExtension

ExE-Boss opened the pull request:
Seeing as this addon hasn't been updated in two years and is open source, I've decided to update it to the new WebExtensions API.

ExE-Boss committed:
Convert to a WebExtension

Signed-off-by: ExE Boss <ExE-Boss@users.noreply.github.com>

Ede123 commented:
Great!
Will review next week or so (currently traveling on holiday).

Do you know if/what has to be done for properly transitioning from 
bootstrapped extension to webeextension? (I.e. automatically update all 
users without hassle)


Am 15. April 2017 6:52:36 nachm. schrieb ExE Boss <notifications@github.com>:

> Seeing as this addon hasn't been updated in two years and is open source, 
> I've decided to update it to the new WebExtensions API.
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Ede123/nicermediapages/pull/1
>
> -- Commit Summary --
>
>   * Convert to a WebExtension
>
> -- File Changes --
>
>     D bootstrap.js (45)
>     M build.sh (2)
>     D chrome.manifest (2)
>     M content/nicermediapages.js (37)
>     D install.rdf (26)
>     A manifest.json (21)
>
> -- Patch Links --
>
> https://github.com/Ede123/nicermediapages/pull/1.patch
> https://github.com/Ede123/nicermediapages/pull/1.diff
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Ede123/nicermediapages/pull/1

ExE-Boss commented:
Simply upload it to AMO as a new version of the current addon at https://addons.mozilla.org/firefox/addon/nicermediapages/ (I have already made sure that the addon ID in `manifest.js` is [correct](https://developer.mozilla.org/Add-ons/WebExtensions/WebExtensions_and_the_Add-on_ID))

**See also:**
https://developer.mozilla.org/Add-ons/WebExtensions/Porting_a_legacy_Firefox_add-on

I also ask that you add a link to my AMO profile at: https://addons.mozilla.org/en-US/firefox/user/ExE_Boss/

ExE-Boss committed:
Handle Nicer Media Pages initialization with the `run_at` manifest key

ExE-Boss committed:
Remove unnecessary permissions

  The "activeTab" permission is unused by this extension, so including
it is a security hole,as it asks for more permissions than is necessary
by this extension.

Ede123 commented:
Alright, looks good to me!
Initial tests look promising. If I don't encounter any issues I'll release a new version soon.

Thank you very much for the contribution!
